src/v/v8_engine/instance.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/core/temporary_buffer.hh>
+
+#include <v8.h>
+
+namespace v8_engine {
+
+class instance_exception final : public std::exception {
+public:
+    explicit instance_exception(ss::sstring msg)
+      : _msg(std::move(msg)) {}
+
+    const char* what() const noexcept final { return _msg.c_str(); }
+
+private:
+    ss::sstring _msg;
+};
+
+// This class implemented container for run js code inside it.
+// It stores only one function from js code and can run it.
+// v8::Isolate are used like container for js env.
+// It can help isolate different functions from each other
+// and set up constraints for them separately.
+class instance {
+public:
+    // Init new instance.
+    // Create isolate and configure constraints for it.
+    explicit instance(size_t max_heap_size_in_bytes);
```

Comment:
For future refactoring: 

I think this should be a `pimpl` idiom 

we should have a more general `inline_xform` or `lambda` or `inlinefunc` 

v8 should be an extrict implementation detail in case we want to add wasmer, or other bytecodealliance engines later.